geoapps/io/driver.py

```diff
@@ -0,0 +1,39 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+import json
+import os
+
+
+class InputFile:
+    def __init__(self, filename):
+        self._accepted_file_extensions = ["json"]
+        self.data = None
+        self.filename = filename
+
+    @property
+    def filename(self):
```

Comment:
```suggestion
    def filename(self) -> str:
```

Reply:
Just noticed these comments now.  I went back over and added typing support, but I didn't think to do it for the setters/getters.